opcua/common/events.py

```diff
@@ -134,6 +145,7 @@ def append_new_attribute_to_select_clauses(attribute, already_selected, parent_b
     else:
         already_selected[string_path] = string_path
         op = ua.SimpleAttributeOperand()
+        op.TypeDefinitionId = NodeId(identifier=ua.ObjectIds.BaseEventType, namespaceidx=0, nodeidtype=NodeIdType.Numeric)
```

Comment:
Without this, the subscription call breaks and returns errors.

Reply:
I thought there were no docs for TypeDefinitionId but it explains a little something in
https://reference.opcfoundation.org/v105/Core/docs/Part4/7.22.3/:
> In some cases the same SimpleAttributeOperand.browsePath will apply to multiple EventTypes. If the Client specifies the BaseEventType in the SimpleAttributeOperand.typeDefinitionId then the Server shall evaluate the SimpleAttributeOperand.browsePath without considering the SimpleAttributeOperand.typeDefinitionId.


My concern here is whether adding in this to the request call would cause problems in environments that were already fine without it. I'm testing that out just to be safe.

Reply:
I say an interesting post about this here: https://stackoverflow.com/questions/62324020/how-to-properly-use-the-typedefinition-and-object-instances-in-opc-ua-with-the-o